addon/mixins/picker.js

```diff
@@ -100,7 +100,7 @@ export default Mixin.create(CancelableMixin, {
   }),
 
   focusOnInput() {
-    let element = document.querySelector("." + this.get('topClass') + " .dp-date-input");
+    let element = this.$(`.${this.get('topClass')} .dp-date-input`).get(0);
```

Comment:
jQuery has a `.first()` which would give the first result wrapped in a jquery object. So you should be able to call `.first().focus().select()` without caring if any element actually matched.